apps/mobile/src/components/ProfileView/ProfileView.tsx

```diff
@@ -181,28 +185,88 @@ type ProfileViewUsernameProps = {
   queryRef: ProfileViewUsernameFragment$key;
 };
 
+const BADGE_DESCRIPTIONS: Record<string, string> = {
+  'Top Member':
+    'Awarded for being one of the most active contributors on Gallery in the past week.',
+  'Community pillar badge': 'Awarded for high-quality engagement',
+};
+
 export function ProfileViewUsername({ queryRef, style }: ProfileViewUsernameProps) {
   const query = useFragment(
     graphql`
       fragment ProfileViewUsernameFragment on Query {
         userByUsername(username: $username) {
           ... on GalleryUser {
             username
+            badges {
+              name
+              imageURL
+            }
           }
         }
+        ...isFeatureEnabledFragment
       }
     `,
     queryRef
   );
 
+  const isActivityBadgeEnabled = isFeatureEnabled(FeatureFlag.ACTIVITY_BADGE, query);
+
+  const bottomSheetRef = useRef<GalleryBottomSheetModalType | null>(null);
+
+  const filteredBadges = useMemo(() => {
+    const badges = query.userByUsername?.badges ?? [];
+    return badges.filter((badge) => badge?.imageURL);
+  }, [query.userByUsername?.badges]);
```

Comment:
i think we want to further by `BADGE_ENABLED_COMMUNITY_ADDRESSES`, similar to other usages

Reply:
Updated the logic just like in web